Fix left-shift with vartabstop issue #14864

ghgary opened the pull request:
"Left shift is incorrect with vartabstop and shiftwidth=0"

The problem was that with 'vartabstop' set and 'shiftwidth' equal 0, left shifts using << were shifting the line to the wrong column.  The tabstop to the right of the first character in the line was being used as the shift amount instead of the tabstop to the left of that first character.

The reason was that the tabstop_at() function always returned the value of the tabstop to the right of the given column and was not accounting for the direction of the shift.

The solution was to make tabstop_at() aware of the direction of the shift and to choose the tabtop accordingly.

A test was added to check this behavior.

Commit:
Fix left-shift with vartabstop issue #14864

"Left shift is incorrect with vartabstop and shiftwidth=0"

The problem was that with 'vartabstop' set and 'shiftwidth' equal 0,
left shifts using << were shifting the line to the wrong column.  The
tabstop to the right of the first character in the line was being used
as the shift amount instead of the tabstop to the left of that first
character.

The reason was that the tabstop_at() function always returned the value
of the tabstop to the right of the given column and was not accounting
for the direction of the shift.

The solution was to make tabstop_at() aware of the direction of the
shift and to choose the tabtop accordingly.

A test was added to check this behavior.

yegappan commented on src/testdir/test_vartabs.vim:
```diff
@@ -454,5 +454,62 @@ func Test_vartabstop_latin1()
+  norm! <<
+  let expect = "word"
+  call assert_equal(expect, getline(1))
+endfunc
```
Can you add a `bw!` at the end of this function?

ghgary replied:
Done.

Commit:
Add :bwipeout! to end of new test function

Commit:
Mark new function parameter UNUSED

The new parameter "left" added to the get_sw_value_col() function is
unused when FEAT_VARTABS is not defined, causing the build to fail.

zeertzjq commented on src/indent.c:
```diff
@@ -123,14 +123,22 @@ tabstop_padding(colnr_T col, int ts_arg, int *vts)
+ *
+ * If this is being called as part of a shift operation, col is not the cursor
+ * column but is the column number to the left of the first non-whitespace
+ * character in the line.  If the shift is to the left (left = 1), then return
```
Should `left` use `TRUE` and `FALSE` instead of `0` and `1`? `TRUE` and `FALSE` are passed to the `left` argument of `shift_line()`, for example in indent.c:
```c
		shift_line(TRUE, FALSE, 1, TRUE);
```

ghgary replied:
I think you're right. Fixed.

Commit:
Use FALSE instead of 0 to indicate not-left

zeertzjq commented on src/indent.c:
```diff
@@ -123,14 +123,22 @@ tabstop_padding(colnr_T col, int ts_arg, int *vts)
+ * If this is being called as part of a shift operation, col is not the cursor
+ * column but is the column number to the left of the first non-whitespace
+ * character in the line.  If the shift is to the left (left = 1), then return
+ * the size of the tab interval to the left of the column.
```
It should also be `TRUE` instead of `1` in this comment
```suggestion
 * character in the line.  If the shift is to the left (left = TRUE), then
 * return the size of the tab interval to the left of the column.
```

ghgary replied:
Yep. Thanks. Done.

Commit:
Change 1 to TRUE in a comment
